src/crimson/os/futurized_store.h

```diff
@@ -24,9 +24,13 @@ class Transaction;
 
 namespace crimson::os {
 class FuturizedCollection;
+class FuturizedShardStore;
+
+constexpr core_id_t PRIMARY_CORE = 0;
```

Comment:
How about deducing the primary core of FuturizedStore in its constructors, thus relax this constraint at the level of futurized_store?

Reply:
done in a new commit.